pkg/catalogv2/git/git.go

```diff
@@ -1,480 +1,318 @@
+func (ro *repoOperation) GetConfig() *git {
+	return ro.config
 }
 
-func (g *git) httpClientWithCreds() (*http.Client, error) {
-	var (
-		username  string
-		password  string
-		tlsConfig tls.Config
-	)
-
-	if g.secret != nil {
-		switch g.secret.Type {
-		case corev1.SecretTypeBasicAuth:
-			username = string(g.secret.Data[corev1.BasicAuthUsernameKey])
-			password = string(g.secret.Data[corev1.BasicAuthPasswordKey])
-		case corev1.SecretTypeTLS:
-			cert, err := tls.X509KeyPair(g.secret.Data[corev1.TLSCertKey], g.secret.Data[corev1.TLSPrivateKeyKey])
-			if err != nil {
-				return nil, err
-			}
-			tlsConfig.Certificates = append(tlsConfig.Certificates, cert)
-		}
-	}
-
-	if len(g.caBundle) > 0 {
-		cert, err := x509.ParseCertificate(g.caBundle)
-		if err != nil {
-			return nil, err
-		}
-		pool, err := x509.SystemCertPool()
-		if err != nil {
-			pool = x509.NewCertPool()
-		}
-		pool.AddCert(cert)
-		tlsConfig.RootCAs = pool
-	}
-
-	transport := http.DefaultTransport.(*http.Transport).Clone()
-	transport.TLSClientConfig = &tlsConfig
-	transport.TLSClientConfig.InsecureSkipVerify = g.insecureTLSVerify
-
-	client := &http.Client{
-		Transport: transport,
-		Timeout:   30 * time.Second,
-	}
-	if username != "" || password != "" {
-		client.Transport = &basicRoundTripper{
-			username: username,
-			password: password,
-			next:     client.Transport,
-		}
-	}
-
-	return client, nil
+// GetAuth just returns the authentication method configured,
+// this method is intended for unit-tests only
+func (ro *repoOperation) GetAuth() transport.AuthMethod {
+	return ro.auth
 }
 
-func (g *git) remoteSHAChanged(branch, sha string) (bool, error) {
-	formattedURL := formatGitURL(g.URL, branch)
-	if formattedURL == "" {
-		return true, nil
-	}
+// plainClone operation of a git repository at given branch with depth = 1
+func (ro *repoOperation) cloneOrOpen(branch string) error {
 
-	client, err := g.httpClientWithCreds()
-	if err != nil {
-		logrus.Warnf("Problem creating http client to check git remote sha of repo [%v]: %v", g.URL, err)
-		return true, nil
+	cloneOptions := ro.cloneOpts
+	if branch != "" {
+		cloneOptions.ReferenceName = plumbing.NewBranchReferenceName(branch)
+		cloneOptions.SingleBranch = true
 	}
-	defer client.CloseIdleConnections()
 
-	req, err := http.NewRequest("GET", formattedURL, nil)
+	err := cloneOptions.Validate()
 	if err != nil {
-		logrus.Warnf("Problem creating request to check git remote sha of repo [%v]: %v", g.URL, err)
-		return true, nil
+		return fmt.Errorf("plainClone validation failure - error: %v", err)
 	}
 
-	req.Header.Set("Accept", "application/vnd.github.v3.sha")
-	req.Header.Set("If-None-Match", fmt.Sprintf("\"%s\"", sha))
-	for k, v := range g.headers {
-		req.Header.Set(k, v)
+	localRepository, err := gogit.PlainClone(ro.config.Directory, false, cloneOptions)
+	if err != nil && err != gogit.ErrRepositoryAlreadyExists {
+		return fmt.Errorf("plainClone failure - error: %v", err)
```

Comment:
Could wrap the error:
```suggestion
		return fmt.Errorf("plainClone failure: %w", err)
```